hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py

```diff
@@ -0,0 +1,171 @@
+    for privacy_group in unique_privacy_groups:
+        if privacy_group.threat_updates_enabled:
+            # HMA can only read from privacy groups that have threat_updates enabled.
+            # # See here for more details:
+            # https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-updates/v9.0
+            logger.info("Adding collaboration name %s", privacy_group.name)
+            priavcy_group_id_in_use.add(privacy_group.id)
+            config = ThreatExchangeConfig(
+                privacy_group.id,
+                # TODO Currently default to True for testing purpose,
+                # need to switch it to False before v0 launch
+                fetcher_active=True,
+                privacy_group_name=privacy_group.name,
+                in_use=True,
+                write_back=False,
+            )
+            # Warning! Will stomp on existing configs (including if you disable them)
+            # TODO need to compare with existing privacy groups in dynamoDB to create/update/delete
+            hmaconfig.update_config(config, insert_only=True)
+    update_privacy_group_in_use(priavcy_group_id_in_use)
+
+
+def update_privacy_group_in_use(priavcy_group_id_in_use: set) -> None:
+    collabs = ThreatExchangeConfig.get_all()
+    for collab in collabs:
+        if collab.privacy_group_id not in priavcy_group_id_in_use:
+            hmaconfig.update_config_attributes_by_type_and_name(
+                config_type="ThreatExchangeConfig",
+                name=str(collab.privacy_group_id),
+                updates={"in_use": False},
+            )
+
+
+def get_datasets_api(hma_config_table: str) -> bottle.Bottle:
+    """
+    A Closure that includes all dependencies that MUST be provided by the root
+    API that this API plugs into. Declare dependencies here, but initialize in
+    the root API alone.
+    """
+
+    # A prefix to all routes must be provided by the api_root app
+    # The documentation below expects prefix to be '/datasets/'
+    datasets_api = bottle.Bottle()
+    HMAConfig.initialize(hma_config_table)
+
+    @datasets_api.get("/", apply=[jsoninator])
+    def datasets() -> DatasetsResponse:
+        """
+        Returns all datasets.
+        """
+        collabs = ThreatExchangeConfig.get_all()
+        return DatasetsResponse(
+            datasets_response=[
+                Dataset(
+                    privacy_group_id=collab.privacy_group_id,
+                    privacy_group_name=collab.privacy_group_name,
+                    fetcher_active=collab.fetcher_active,
+                    write_back=collab.write_back,
+                    in_use=collab.in_use,
+                )
+                for collab in collabs
+            ]
+        )
+
+    @datasets_api.put("/update", apply=[jsoninator])
+    def update_datasets():
+        """
+        Update dataset
+        """
+        dataset = bottle.request.json
+        updates = {}
+        if dataset["fetcher_active"] is not None:
```

Comment:
Untyped request json is also hard to debug / understand / maintain. Consider adding an extension to jsoninator which does typed json processing and adds that as an argument to the view method.

Something like 
```python
@dataclass
class RequestType():
  <define types here>

@datasets_api.put("/update", apply=[jsoninator(RequestType)])
def update_datasets(typed_payload: RequestType):
  do_something()
```